docs/source/jit.rst

```diff
@@ -471,6 +470,31 @@ Example (a type mismatch)
      ...
 
 
+Unsupported Types
+^^^^^^^^^^^^^^^^^
+Some parts of the :mod:`typing` module are missing from TorchScript. Some of these
+are more fundamental things that are unlikely to be added in the future while others
+may be added if there is enough user demand to make it a priority.
+
+These types and features from the :mod:`typing` module are unavailble in TorchScript.
+
+.. csv-table::
+   :header: "Item", "Description"
+
+   ":any:`typing.Any`", ":any:`typing.Any` is currently in development but not yet released"
+   ":any:`typing.NoReturn`", "Not implemented"
+   ":any:`typing.Union`", "Unlikely to be implemented"
```

Comment:
note that we do support Optionals

Reply:
I added a note, but it's also listed above in the supported types